tests/run/breaks.scala

```diff
@@ -0,0 +1,38 @@
+import scala.util.*
+import collection.mutable.ListBuffer
+
+object Test {
+  def has(xs: List[Int], elem: Int) =
+    boundary:
+      for x <- xs do
+        if x == elem then break(true)
+      false
+
+  def takeUntil(xs: List[Int], elem: Int) =
+    boundary:
+      var buf = new ListBuffer[Int]
+      for x <- xs yield
+        if x == elem then break(buf.toList)
+        buf += x
+        x
```

Comment:
This is not the implementation I would expect. It builds *2* result lists and only uses one of them. I would expect the following instead:
```suggestion
      for x <- xs do
        if x == elem then break(buf.toList)
        buf += x
      xs
```